### PR TITLE
Eta lifts function before args.

### LINKED ISSUE #11465: Evaluation of subexpressions reversed under eta expansion
No one complained about this, which is described this way in the spec example:
```
scala 2.13.0-M5> def f = { println("F") ; 1 to 10 }
f: scala.collection.immutable.Range.Inclusive

scala 2.13.0-M5> f.foldLeft{ println("Z"); 0 }(_ + _)
F
Z
res0: Int = 55

scala 2.13.0-M5> f.foldLeft{ println("Z"); 0 } _
Z
F
res1: ((Int, Int) => Int) => Int = $$Lambda$954/0x0000000800623040@6f50d55c
```
Requisite "works in dotty":
```
Starting dotty REPL...
scala> def f = { println("F") ; 1 to 10 }                                       
def f: scala.collection.immutable.Range.Inclusive

scala> f.foldLeft{ println("Z"); 0 }(_ + _)                                     
F
Z
val res0: Int = 55

scala> f.foldLeft{ println("Z"); 0 } _                                          
F
Z
val res1: ((Int, Int) => Int) => Int = Lambda$1379/0x0000000802028040@56cfe6be

scala>
```

### PR DESCRIPTION
**Pull request by hrhino**
The comment says it, the spec says it, and most importantly, Dotty says it.

Maybe qualify for the coveted "performance" label by avoiding an argument load.

Fixes scala/bug#11465.

### CODE REVIEW

File: test/files/run/t11465.check [hunk 1]
```diff
@@ -0,0 +1,6 @@
+F
+Z
+55
+F
+Z
+55
```
Comment (som-snytt) [inline test/files/run/t11465.check:6]: Is `55` some sort of senior citizens' joke?

**Review by som-snytt — CHANGES_REQUESTED** [review]
That was kind of easy. This must also change the examples in the spec 6.7.

**Comment by hrhino** [comment]
Ach, good catch. I just read that bit, to be sure of myself, but I didn't read the examples, because who does that anymore?

**Review by diesalbla — APPROVED** [review]

**Review by som-snytt — APPROVED** [review]
Thanks. When I "contributed" the examples to document the state of affairs, Adriaan was like what are you sure. I could've fixed it back then. Obviously, no one seemed to care. And what does that say?

**Comment by diesalbla** [comment]
@hrhino Is this ready?

**Comment by hrhino** [comment]
But for timing, should be. I'm in no rush, though.

**Review by adriaanm — APPROVED** [review]

**Pull request merged**